database/postgres/postgres.go

```diff
@@ -214,7 +215,7 @@ func (p *Postgres) Close() error {
 
 // https://www.postgresql.org/docs/9.6/static/explicit-locking.html#ADVISORY-LOCKS
 func (p *Postgres) Lock() error {
-	if p.isLocked {
+	if p.isLocked.Load() {
```

Comment:
Same here about fixing the local lock optimization implementation

Reply:
fixed it